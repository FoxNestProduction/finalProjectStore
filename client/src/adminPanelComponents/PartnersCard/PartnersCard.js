import { Box, Card, CardHeader, Button, CardMedia, Typography } from '@mui/material';
import React from 'react';
import PropTypes from 'prop-types';

const PartnersCard = ({ title, url, enabled }) => {
  return (
    <Box>
      <Card
        sx={{
          maxWidth: '1083px',
          maxHeight: '611px',
          width: '100%',
          padding: '10px 24px 24px 24px',
        }}
      >
        <CardHeader
          sx={{
            p: '0 16px 16px 16px',
            position: 'relative',
          }}
          title={(
            <Typography
              sx={{
                fontSize: {
                  mobile: '38px',
                  fontWeight: '500',
                  fontFamily: 'fontPoppins',
                },
              }}
            >
              {title}
            </Typography>
          )}
          action={(
            <Button
              onClick={(event) => {
                event.preventDefault();
              }}
              aria-label="disable"
              variant="outlined"
              sx={{
                border: 'none',
                outline: '2px solid #FB471D',
                color: '#000000',
                fontFamily: 'fontFamily',
                fontSize: {
                  mobile: '18px',
                },
                px: '1.5em',
                position: 'absolute',
                top: '14px',
                right: '0',
                transition: 'background-color 0.3s ease, box-shadow 0.3s ease',
                '&:hover': {
                  boxShadow: '0px 0px 10px rgba(0, 0, 0, 0.3)',
                  backgroundColor: 'common.white',
                  outline: 'none',
                  border: 'none',
                },
                '&:active': {
                  boxShadow: '0px 1px 5px rgba(0, 0, 0, 0.2)',
                  transform: 'translateY(1px)',
                  backgroundColor: '#EAEAEA',
                },
              }}
            >
              Disable
            </Button>
          )}
        />
        <Box
          sx={{
            position: 'relative',
          }}
        >
          {!enabled ? (
            <Box
              sx={{
                position: 'absolute',
                top: '0',
                left: '0',
                width: '100%',
                height: '100%',
                background: 'rgba(217, 217, 217, 0.60)',

              }}
            />
          ) : null}
          <CardMedia
            component="img"
            image={`${url}`}
            sx={{
              display: 'block',
              width: '1035px',
              height: '495px',
              borderRadius: '16px',
            }}
            alt={`Restaurant ${title}`}
          />
        </Box>
      </Card>
    </Box>
  );
};

PartnersCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string,
  enabled: PropTypes.bool.isRequired,
};

PartnersCard.defaultProps = {
  url: '',
};

export default PartnersCard;
