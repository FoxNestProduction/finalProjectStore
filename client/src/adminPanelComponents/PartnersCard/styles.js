export const card = {
  maxWidth: '1083px',
  maxHeight: '611px',
  width: '100%',
  padding: '10px 24px 24px 24px',
};
export const cardHeader = {
  p: {
    tablet: '0 0 16px',
    lgTablet: '0 16px 16px 16px',
  },
  position: 'relative',
};

export const cardTitle = {
  fontSize: {
    mobile: '26px',
    tablet: '28px',
    lgTablet: '30px',
    desktop: '38px',
  },
  fontWeight: '500',
  fontFamily: 'fontPoppins',
};

export const disableButton = {
  border: 'none',
  outline: '2px solid #FB471D',
  color: '#000000',
  fontFamily: 'fontFamily',
  fontSize: {
    mobile: '18px',
  },
  px: {
    tablet: '',
    lgTablet: '1em',
    desktop: '1.5em',
  },
  position: {
    tablet: '',
    lgTablet: 'absolute',
    desktop: 'absolute',
  },
  top: '14px',
  right: '0',
  transition: 'background-color 0.3s ease, box-shadow 0.3s ease',
  '&:hover': {
    boxShadow: '0px 0px 10px rgba(0, 0, 0, 0.3)',
    backgroundColor: 'common.white',
    outline: 'none',
    border: 'none',
  },
  '&:active': {
    boxShadow: '0px 1px 5px rgba(0, 0, 0, 0.2)',
    transform: 'translateY(1px)',
    backgroundColor: '#EAEAEA',
  },
};

export const imgWrapper = {
  position: 'relative',
};

export const disabledCardStyles = {
  position: 'absolute',
  top: '0',
  left: '0',
  width: '100%',
  height: '100%',
  background: 'rgba(217, 217, 217, 0.60)',
  borderRadius: '16px',
};

export const imgStyles = {
  display: 'block',
  width: '1035px',
  height: '495px',
  borderRadius: '16px',
};
