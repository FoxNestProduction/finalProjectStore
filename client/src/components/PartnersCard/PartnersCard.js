import React from 'react';
import PropTypes from 'prop-types';
import Container from '@mui/material/Container';
import { fixedDecodeURIComponent } from '../../utils/uriEncodeHelpers';
import RestaurantCard from '../RestaurantCard/RestaurantCard';
import useGetPartners from '../../customHooks/useGetPartners';

const PartnersCard = ({ partnersName }) => {
  const partners = useGetPartners();

  const nameOfPartners = fixedDecodeURIComponent(partnersName);
  const rest = partners.find((item) => item.name.toLowerCase() === nameOfPartners);
  const styleRestaurantCard = {
    mobile: 315,
    lgTablet: 881,
    desktop: 1285,
  };

  return (
    <Container
      component="section"
      sx={{
        bgcolor: 'background.default',
        mt: { mobile: 5, tablet: 8 },
        mb: { mobile: 5, tablet: 8 },
      }}
    >

      <RestaurantCard {...rest} styleWidth={styleRestaurantCard} />
    </Container>
  );
};

PartnersCard.propTypes = {
  partnersName: PropTypes.string.isRequired,
};

export default PartnersCard;
