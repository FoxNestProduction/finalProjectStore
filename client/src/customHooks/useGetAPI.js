import React, { useEffect, useState } from 'react';
import { instance } from '../API/instance';

const useGetAPI = (url) => {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    (async () => {
      try {
        // setLoading(true);
        const response = await instance.get(url);
        setData(response.data);
      } catch (err) {
        setError(err.response);
      } finally {
        setLoading(false);
      }
    })();
  }, [url]);

  return [data, loading, error];
};

export default useGetAPI;
