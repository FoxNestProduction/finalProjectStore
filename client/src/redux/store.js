import { configureStore, getDefaultMiddleware, combineReducers } from '@reduxjs/toolkit';
// todo: правила для redux-persist
// eslint-disable-next-line import/no-extraneous-dependencies
import storage from 'redux-persist/lib/storage';
// eslint-disable-next-line import/no-extraneous-dependencies
import { persistReducer } from 'redux-persist';
import modalSlice from './slices/modalSlice';
import productsSlice from './slices/productsSlice';
import authorizationSlice from './slices/authorizationSlice';
import userSlice from './slices/userSlice';
import partnersSlice from './slices/partnersSlice';
import searchSlice from './slices/searchSlice';
import errorSlice from './slices/errorSlice';
import reviewsSlice from './slices/reviewsSlice';
import favouriteSlice from './slices/favouriteSlice';

const authPersistConfig = {
  key: 'authorization',
  version: 1,
  storage,
};

const userPersistConfig = {
  key: 'user',
  version: 1,
  storage,
};

const searchPersistConfig = {
  key: 'search',
  version: 1,
  storage,
};
const favouritePersistConfig = {
  key: 'favorites',
  version: 1,
  storage,
};

/*
    todo: - ЗВЕРНІТЬ УВАГУ!
      якщо вам потрібно щоб ваш стейт не зникав після перезавантаження
      то юзайте підхід як authorization і user,
      якщо ж ні то варіант написання як для modal всередині reducer
      Якщо що пишіть @Ihor_Kacher
*/
const reducer = combineReducers({
  error: errorSlice,
  modal: modalSlice,
  partners: partnersSlice,
  authorization: persistReducer(authPersistConfig, authorizationSlice),
  user: persistReducer(userPersistConfig, userSlice),
  products: productsSlice,
  search: persistReducer(searchPersistConfig, searchSlice),
  reviews: reviewsSlice,
  favourites: persistReducer(favouritePersistConfig, favouriteSlice),
});

const middleware = getDefaultMiddleware({
  serializableCheck: false,
});

const store = configureStore({
  reducer,
  middleware,
});

export default store;
