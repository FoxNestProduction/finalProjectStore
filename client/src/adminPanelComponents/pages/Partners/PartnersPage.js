import { Box, Typography, Container, Stack, Autocomplete, TextField, InputAdornment, Button } from '@mui/material';
import React, { useEffect, useRef } from 'react';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import SearchIcon from '@mui/icons-material/Search';
import { Link, useNavigate } from 'react-router-dom';
import AddBoxOutlinedIcon from '@mui/icons-material/AddBoxOutlined';
import { fetchAllPartnersNames } from '../../../redux/slices/partnersSlice';
import {
  stylesBorder,
  stylesSearch,
  container,
  allContentWrapper,
  searchButtonsTitleWrapper,
  searchAndButtonsWrapper,
  searchBtn,
  addPartnersBtn,
  title,
  cardContainer,
} from './styles';
import useGetAPI from '../../../customHooks/useGetAPI';
import PartnersCard from '../../PartnersCard/PartnersCard';
import { outlinedBtnStyles, containedBtnStyles } from '../../../muiTheme/buttonsStyles';
import { setInputSearchValue, setSearch, fetchSearchedProductsOrPartners } from '../../../redux/slices/searchSlice';

const PartnersPage = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  useEffect(() => {
    dispatch(fetchAllPartnersNames());
  }, [dispatch]);
  const [partners, loading] = useGetAPI('/partners');

  const handleAddNewPartners = () => {
    navigate('/admin-panel/partners/new-partner');
  };

  const itemsFromSearch = useSelector((state) => state.search.search, shallowEqual);
  const allPartnersNames = useSelector((state) => state.partners.allPartnersNames, shallowEqual);
  const inputSearchValue = useSelector((state) => state.search.inputSearchValue);
  /* Варіант коду з пошуком через сервер
 const handleSearchFetch = (value) => {
    if (value === '') {
      dispatch(setSearch([]));
    } else {
      const fetchData = {
        url: '/partners/search',
        body: {
          query: inputSearchValue,
        },
      };
      dispatch(fetchSearchedProductsOrPartners(fetchData));
      navigate('');
    }
  };

   const handleInputChange = async (event, newInputValue, reason) => {
    dispatch(setInputSearchValue(newInputValue));
    if (reason === 'clear') {
      handleSearchFetch(newInputValue);
    }
  };

    const handleSearch = (newInputValue) => {
    dispatch(setInputSearchValue(newInputValue));
    handleSearchFetch(newInputValue);
  };

  */
  // Варіант коду з фільтрацією на стороні клієнта
  const handleInputChange = async (event, newInputValue, reason) => {
    dispatch(setInputSearchValue(newInputValue));
    if (reason === 'clear') {
      dispatch(setSearch([]));
    }
  };

  const handleSearch = (newInputValue) => {
    const filteredItem = partners.filter(({ name }) => {
      return inputSearchValue === name || name.includes(inputSearchValue);
    });
    dispatch(setSearch(filteredItem));
  };

  return (
    <Container
      sx={container}
    >
      <Box
        sx={allContentWrapper}
      >
        <Box
          sx={searchButtonsTitleWrapper}
        >
          <Box
            sx={searchAndButtonsWrapper}
          >
            <Stack
              sx={stylesSearch}
            >
              <Autocomplete
                inputValue={inputSearchValue}
                options={allPartnersNames}
                onInputChange={handleInputChange}
                id="search-partners"
                freeSolo
                blurOnSelect
                renderInput={(params) => (
                  <TextField
                    sx={{
                      ...stylesBorder,
                      '& .MuiInputBase-input[type="search"]::-webkit-search-cancel-button': {
                        display: 'none',
                      },
                    }}
                    {...params}
                    label="Search Partners"
                    variant="outlined"
                    InputProps={{
                      ...params.InputProps,
                      type: 'search',
                      startAdornment: (
                        <InputAdornment position="start">
                          <SearchIcon />
                        </InputAdornment>
                      ),
                    }}
                  />
                )}
              />
            </Stack>
            <Button
              variant="contained"
              onClick={handleSearch}
              sx={{
                ...containedBtnStyles,
                ...searchBtn,
              }}
            >
              Search
            </Button>
            <Button
              variant="outlined"
              endIcon={<AddBoxOutlinedIcon />}
              onClick={handleAddNewPartners}
              sx={{
                ...outlinedBtnStyles,
                ...addPartnersBtn,
              }}
            >
              Add Partners
            </Button>
          </Box>

          <Typography
            variant="h1"
            sx={title}
          >
            Partners
          </Typography>
        </Box>
        <Box
          sx={cardContainer}
        >
          {
            itemsFromSearch.length !== 0
              ? (itemsFromSearch.map(({ imageUrl, enabled, _id, name, customId }) => (
                <Link
                  key={_id}
                  to={`/admin-panel/partners/${customId}`}
                >
                  <PartnersCard
                    title={name}
                    enabled={enabled}
                    url={imageUrl}
                  />
                </Link>
              )))
              : (partners !== null && partners.length !== 0 && partners.map(
                ({ imageUrl, enabled, _id, name, customId }) => (
                  <Link
                    key={_id}
                    to={`/admin-panel/partners/${customId}`}
                  >
                    <PartnersCard
                      title={name}
                      enabled={enabled}
                      url={imageUrl}
                    />
                  </Link>
                ),
              ))
          }
        </Box>
      </Box>
    </Container>
  );
};

export default PartnersPage;
