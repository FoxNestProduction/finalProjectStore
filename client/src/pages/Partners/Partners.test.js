import React, { Component } from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import PartnersPage from './Partners';
import PartnersCard from '../../components/PartnersCard/PartnersCard';
import useGetAPI from '../../customHooks/useGetAPI';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

jest.mock('../../customHooks/useGetAPI', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('@mui/material/', () => ({
  ...jest.requireActual('@mui/material/'),
  useMediaQuery: jest.fn(),
}));

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useParams: () => ({ name: 'PizzaDay', customId: '321' }),
}));
const partner = {
  rating: 5,
  name: 'PizzaDay',
  imageUrl: 'image1.jpg',
  description: 'PizzaDayAbout',
  customId: '321',
};

describe('PartnersPage Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should renders PartnersPage component', async () => {
    useGetAPI.mockReturnValueOnce([{
      rating: 5,
      name: 'PizzaDay',
      imageUrl: 'image1.jpg',
      description: 'PizzaDayAbout',
      customId: '321',
    }, false, null]);
    useGetAPI.mockReturnValueOnce([null, true, null]);

    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/partners/:321']}>
          <Routes>
            <Route
              path="/restaurants/:name/:customId"
              element={(
                <PartnersPage>
                  <PartnersCard partner={partner} />
                </PartnersPage>
            )}
            />
          </Routes>
        </MemoryRouter>
      </Provider>,
    );

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  test('should render skeleton while loading partner', () => {
    useGetAPI.mockReturnValueOnce([null, true, null]);
    useGetAPI.mockReturnValueOnce([null, true, null]);

    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/restaurants/:321']}>
          <Routes>
            <Route path="/restaurants/:customId" element={<PartnersPage />} />
          </Routes>
        </MemoryRouter>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
