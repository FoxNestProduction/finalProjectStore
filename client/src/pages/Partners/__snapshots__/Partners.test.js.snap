// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartnersPage Component should render skeleton while loading partner 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <section
      class="MuiContainer-root MuiContainer-maxWidthLg css-2zxudu-MuiContainer-root"
    >
      <div
        class="container"
      >
        <div
          class="oneRestaurantPageCardSkeleton"
        >
          <div
            class="restaurantsPageImageSkeleton loading"
          />
          <div
            class="restaurantsPageChipsTitleRatingSkeleton loading"
          />
          <div
            class="restaurantsPageChipsTitleRatingSkeleton loading"
          />
          <div
            class="restaurantsPageShowMoreSkeleton loading"
          />
        </div>
      </div>
    </section>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-sjcq8o-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-1f0n1hj"
      >
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="container"
          >
            <div
              class="productCardSkeleton"
            >
              <div
                class="productImageSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productRaitingSkeleton loading"
              />
              <div
                class="productPriceSkeleton loading"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="container"
          >
            <div
              class="productCardSkeleton"
            >
              <div
                class="productImageSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productRaitingSkeleton loading"
              />
              <div
                class="productPriceSkeleton loading"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="container"
          >
            <div
              class="productCardSkeleton"
            >
              <div
                class="productImageSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productRaitingSkeleton loading"
              />
              <div
                class="productPriceSkeleton loading"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="container"
          >
            <div
              class="productCardSkeleton"
            >
              <div
                class="productImageSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productRaitingSkeleton loading"
              />
              <div
                class="productPriceSkeleton loading"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="container"
          >
            <div
              class="productCardSkeleton"
            >
              <div
                class="productImageSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productRaitingSkeleton loading"
              />
              <div
                class="productPriceSkeleton loading"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="container"
          >
            <div
              class="productCardSkeleton"
            >
              <div
                class="productImageSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productChipsTitleRatingSkeleton loading"
              />
              <div
                class="productRaitingSkeleton loading"
              />
              <div
                class="productPriceSkeleton loading"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <section
      class="MuiContainer-root MuiContainer-maxWidthLg css-83rf2v-MuiContainer-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h5 css-2mz86v-MuiTypography-root"
      >
        Frequently Asked
        <br />
        Questions
      </h3>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How long does delivery take?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How Does It Work?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  How does your food delivery service work?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  What payment options do you accept?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
        >
          <ul
            class="MuiList-root css-1mk9mw3-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters css-1wxbwxm-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h3 css-krdrhq-MuiTypography-root"
                >
                  Do you offer discounts or promotions?
                </h5>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  edge="end"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
        </li>
      </ul>
    </section>
  </div>
</DocumentFragment>
`;

exports[`PartnersPage Component should renders PartnersPage component 1`] = `<DocumentFragment />`;
